spec/forms/contact_phone_forms_spec.rb

```diff
@@ -13,37 +18,90 @@
       let(:contact_phone_form) { build(:contact_phone_form, :has_required_data) }
-      let(:invalid_params) { { reg_identifier: "foo" } }
+      let(:invalid_params) do
+        {
+          reg_identifier: "foo",
+          phone_number: "foo"
+        }
+      end
 
       it "should not submit" do
         expect(contact_phone_form.submit(invalid_params)).to eq(false)
       end
     end
   end
 
-  describe "#reg_identifier" do
-    context "when a valid transient registration exists" do
-      let(:transient_registration) do
-        create(:transient_registration,
-               :has_required_data,
-               workflow_state: "contact_phone_form")
-      end
-      # Don't use FactoryBot for this as we need to make sure it initializes with a specific object
-      let(:contact_phone_form) { ContactPhoneForm.new(transient_registration) }
+  context "when a valid transient registration exists" do
+    let(:contact_phone_form) { build(:contact_phone_form, :has_required_data) }
 
+    describe "#reg_identifier" do
       context "when a reg_identifier meets the requirements" do
+        it "is valid" do
+          expect(contact_phone_form).to be_valid
+        end
+      end
+
+      context "when a reg_identifier is blank" do
         before(:each) do
-          contact_phone_form.reg_identifier = transient_registration.reg_identifier
+          contact_phone_form.reg_identifier = ""
         end
 
+        it "is not valid" do
+          expect(contact_phone_form).to_not be_valid
+        end
+      end
+    end
+
+    describe "#phone_number" do
+      context "when a phone_number meets the requirements" do
         it "is valid" do
           expect(contact_phone_form).to be_valid
         end
       end
 
-      context "when a reg_identifier is blank" do
+      context "when a UK phone_number is formatted to include +44" do
         before(:each) do
-          contact_phone_form.reg_identifier = ""
+          contact_phone_form.phone_number = "+44 1234 567890"
+        end
+
+        it "is valid" do
+          expect(contact_phone_form).to be_valid
+        end
+      end
+
+      context "when a phone_number is a valid international number" do
+        before(:each) do
+          contact_phone_form.phone_number = "+1-202-555-0109"
+        end
+
+        it "is valid" do
+          expect(contact_phone_form).to be_valid
+        end
+      end
+
+      context "when a phone_number is blank" do
+        before(:each) do
+          contact_phone_form.phone_number = ""
+        end
+
+        it "is not valid" do
+          expect(contact_phone_form).to_not be_valid
+        end
+      end
+
+      context "when a phone_number is too long" do
+        before(:each) do
+          contact_phone_form.phone_number = "01234 567 890 123"
+        end
+
+        it "is not valid" do
+          expect(contact_phone_form).to_not be_valid
+        end
+      end
+
+      context "when a phone_number is not a valid number" do
```

Comment:
Because of how 'clever' the phonelib library can be, and coming at specs as a form of documentation, I would suggest being a little clearer with the contexts e.g.

```ruby
context "when phone_number is is not a number" do
  before(:each) do
    contact_phone_form.phone_number = "foo"
  end
end
context "when phone_number is not a valid number" do
  before(:each) do
    # It might look valid, but actually this is not a recognised number
    contact_phone_form.phone_number = "0117 785 3149"
  end
end
```

Reply:
Sounds like a good idea to me! A good way to illustrate what the gem is actually doing. Will make changes now.